src/util/optional.h

```diff
@@ -6,23 +6,25 @@
 
 #include "cereal/cereal.hpp"
 #include <memory>
+#include <optional>
```

Comment:
any reason to keep this file? This was our poor man's implementation of `std::optional`

Reply:
removing